tests/standard/test_file_source.py

```diff
@@ -0,0 +1,62 @@
+import dlt
+import pytest
+
+from sources.standard.filesystem import filesystem_resource
+from tests.utils import ALL_DESTINATIONS, assert_load_info, load_table_counts
+
+
+@pytest.mark.parametrize("destination_name", ALL_DESTINATIONS)
+def test_all_resources(destination_name: str) -> None:
```

Comment:
we need a test that does not load data to database - you just instantiate `filesystem` with many different gobs and buckets and iterate it via list(). all resources and sources are just iterators. the tests I see:
- no glob - all files are returned (recursively)
- recursive and non recursive globs
- glob that is parsing files that contain dates (ie YYYY-MM-DD). - that one of the use cases
- you need to test all the details of the file item! (path, url, sizes, mtimes etc.) - I know this is some work so maybe you can test just one file (per bucket) for all the settings

Reply:
also test the FIleSystemDict operations
* load all the files
* open all the files (binary and text if applicable)
* verify the content of a single selected file